### PR TITLE
07001 queue thread idle callback

### LINKED ISSUE #7001: Add idle callback to queue thread
When a queue thread is sitting idle long enough, it would be nice to have the ability to invoke a callback.

### PR DESCRIPTION
**Pull request by cody-littley**
Closes #7001

### CODE REVIEW

File: platform-sdk/swirlds-unit-tests/common/swirlds-common-test/src/test/java/com/swirlds/common/test/threading/QueueThreadTests.java [hunk 1]
```diff
@@ -976,4 +976,84 @@ void waitUntilNotBusyTest() throws InterruptedException {
...
+            try {
+                latch3.await();
+            } catch (final InterruptedException ignored) {
+                error.set(true);
+                Thread.currentThread().interrupt();
+            }
+        });
+        queue.start();
+
+        // The queue should be call the idle callback during this time,
```
Comment (alittley) [inline platform-sdk/swirlds-unit-tests/common/swirlds-common-test/src/test/java/com/swirlds/common/test/threading/QueueThreadTests.java:1032]: grammar for this and the other matching comments:

"The queue should ~~be~~ call the idle callback..."

**Review by alittley — DISMISSED** [review]

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/threading/framework/internal/AbstractQueueThreadConfiguration.java [hunk 2]
```diff
@@ -58,6 +62,16 @@
 
     private QueueThreadMetricsConfiguration metricsConfiguration;
 
+    /**
+     * The callback to run periodically when the queue is idle.
+     */
+    private Runnable idleCallback;
```
Comment (lpetrovic05) [inline platform-sdk/swirlds-common/src/main/java/com/swirlds/common/threading/framework/internal/AbstractQueueThreadConfiguration.java:68]: this should probably be an `InterruptableRunnable`

**Review by lpetrovic05 — APPROVED** [review]

**Review by hendrikebbers — APPROVED** [review]

File: platform-sdk/swirlds-unit-tests/common/swirlds-common-test/src/test/java/com/swirlds/common/test/threading/QueueThreadTests.java (hunk 1 above)
Comment (cody-littley) [reply platform-sdk/swirlds-unit-tests/common/swirlds-common-test/src/test/java/com/swirlds/common/test/threading/QueueThreadTests.java:1032]: fixed

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/threading/framework/internal/AbstractQueueThreadConfiguration.java (hunk 2 above)
Comment (cody-littley) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/threading/framework/internal/AbstractQueueThreadConfiguration.java:68]: good idea, done

**Review by alittley — DISMISSED** [review]

**Review by lpetrovic05 — APPROVED** [review]

**Pull request merged**